pkg/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -723,14 +734,63 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 	return nil
 }
 
+func (c *controller) updateProgress(pvc *corev1.PersistentVolumeClaim, pod *corev1.Pod, populatorName string, cr *unstructured.Unstructured) error {
+	var disk_re = regexp.MustCompile(fmt.Sprintf(`volume_populators_%s\{image_id=\"([a-z]|\d|\-)+"\} (\d{1,3}\.?(\d+e\+\d+)*)`, populatorName))
+	url := fmt.Sprintf("http://%s:2112/metrics", pod.Status.PodIP)
+	resp, err := http.Get(url)
+	if err != nil {
+		if strings.Contains(err.Error(), "connection refused") {
+			return nil
+		}
+		c.recorder.Eventf(pod, corev1.EventTypeWarning, reasonPopulatorProgress, "Failed to get url: %s, err: %w", url, err)
+
+		return nil
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		c.recorder.Eventf(pod, corev1.EventTypeWarning, reasonPopulatorProgress, "Failed to read bosy: %s, err: %w", url, err)
+		return nil
+	}
+
+	matches := disk_re.FindAllStringSubmatch(string(body), -1)
+	if matches == nil {
+		c.recorder.Eventf(pod, corev1.EventTypeWarning, reasonPopulatorProgress, "Failed to find matchs: %s", matches)
+		return nil
+	}
+
+	for _, match := range matches {
+		imageID := match[1]
+		progress, _ := strconv.ParseFloat(string(match[2]), 64)
+		c.recorder.Eventf(pod, corev1.EventTypeNormal, reasonPopulatorProgress, "disk_nr: %s, progress %d", imageID, int64(progress))
+		gvr := schema.GroupVersionResource{Group: *pvc.Spec.DataSourceRef.APIGroup, Version: "v1beta1", Resource: "openstackvolumepopulators"}
+		openstackPopulator := &v1beta1.OpenstackVolumePopulator{}
+		runtime.DefaultUnstructuredConverter.FromUnstructured(cr.UnstructuredContent(), openstackPopulator)
+		openstackPopulator.Status.Transferred = fmt.Sprintf("%d", int64(progress))
+		unstructuredPopulator, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(openstackPopulator)
+		cr, err := c.dynamicClient.Resource(gvr).Namespace(pvc.Namespace).Update(context.TODO(), &unstructured.Unstructured{Object: unstructuredPopulator}, metav1.UpdateOptions{})
+		if err != nil {
+			c.recorder.Eventf(pod, corev1.EventTypeWarning, reasonPopulatorProgress, "failed to find cr: %w", err)
+			return nil
+		}
+		c.recorder.Eventf(pod, corev1.EventTypeWarning, reasonPopulatorProgress, "CR: %v", cr)
+
+	}
+
+	return nil
+}
+
 func makePopulatePodSpec(pvcPrimeName, secretName string) corev1.PodSpec {
 	nonRoot := true
 	allowPrivilageEscalation := false
 	user := int64(qemuGroup)
 	return corev1.PodSpec{
 		Containers: []corev1.Container{
 			{
-				Name: populatorContainerName,
+				Name:            populatorContainerName,
+				ImagePullPolicy: corev1.PullAlways,
```

Comment:
better to drop the image pull policy

Reply:
it's for debugging, must not forget :)

Reply:
but that's the beauty of the default policy - that when the "latest" tag is used, it always pull the image - that's handy during development (we use 'devel' tag by default in our script, we can change it to 'latest')